src/core/libmaven/PeakGroup.cpp

```diff
@@ -1069,3 +1059,91 @@ void PeakGroup::setGroupId(int groupId)
         peak.groupNum = groupId;
     }
 }
+void PeakGroup::setUserLabel(char label)
+{
+    if (label == 'g') {
+        _predictedLabel = ClassifiedLabel::Signal;
+    } else if (label == 'b') {
+        _predictedLabel = ClassifiedLabel::Noise;
+    } else if (label == '\0') {
+        _predictedLabel = ClassifiedLabel::None;
+    } else {
+        // ignore invalid labels
+        return;
+    }
+    _predictionProbability = 0.0f;
+    _predictionInference.clear();
+    _userLabel = label;
+
+    if (parent != nullptr && parent->tagString == C12_PARENT_LABEL) {
+        parent->setUserLabel(label);
+        return;
+    }
+
+    for (auto& child : _childIsotopes) {
+            child->setUserLabel(label);
+    }
+}
+
+void PeakGroup::setPredictedLabel(const ClassifiedLabel label,
+                                  const float probability)
+{
+    _predictedLabel = label;
+    if (_predictedLabel == ClassifiedLabel::None) {
+        this->_userLabel = '\0';
+    } else if (_predictedLabel == ClassifiedLabel::MaybeGood) {
+        this->_userLabel = '\0';
+    } else if (_predictedLabel == ClassifiedLabel::Noise) {
+        this->_userLabel = 'b';
+    } else {
+        this->_userLabel = 'g';
+    }
+    _predictionProbability = probability;
+
+    if (parent != nullptr && parent->tagString == C12_PARENT_LABEL) {
+        parent->setPredictedLabel(label, probability);
+        return;
+    }
+    for (auto& child : _childIsotopes) {
+        if (child->_predictedLabel != label
+            && child->_predictionProbability != probability)
+            child->setPredictedLabel(label, probability);
+    }
```

Comment:
Same thing here as above. Only difference is, the model is now marking peak-groups instead of a user. The behavior should be consistent in my opinion.

Reply:
Yes, The child peak groups and child adducts are now classified separately. So, we would be requiring this piece of code. Removing this.